### DISCUSSION TITLE
多字幕支持，默认字幕改为ASS或SRT

### DISCUSSION THREAD

**Discussion by wc7086** [spencerwooo/onedrive-vercel-index#468] [Ideas]
VTT太折磨了，很多字幕组的字幕都是在相同的时间轴里加参与者信息，或者同一个场景下需要翻译多行，ASS转成SRT起码能多行显示，VTT直接把部分字母内容挤没了。

https://github.com/DIYgod/DPlayer/issues/1173

**Comment by spencerwooo** [comment]
> The `<track>` HTML element is used as a child of the media elements, `<audio>` and `<video>`. It lets you specify timed text tracks (or time-based data), for example to automatically handle subtitles. The tracks are formatted in WebVTT format (`.vtt` files) — Web Video Text Tracks. - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track

WebVTT 是 HTML5 字幕标准，对于其他格式的字幕支持超出了本项目的功能范围，不会考虑。

**Comment by DogTorrent** [comment]
我搓了个pr(#623 )，默认加载srt/ass/vtt，不过代码比较烂不知道能不能被合，你可以先拉取试试。
实际上也是转的vtt，但是效果好一些，起码同时间字幕不会挤没而是显示为多行。缺点是遇到特殊情况可能就会变成超多行（比如特效字幕或者用叠加做效果的字幕），而且因为要先下载下来字幕所以加载速度比较慢。

ass.js看起来不错，不知道会不会有大佬做

**Reply by wc7086** [reply]
那天我发现vtt是有特效的，ass转vtt应该可以保留不少特效，但是找半天都没找到现成的库，关于vtt特效的描述都只有[官方文档][1]写了，太蛋疼了就没继续深入了。: (

[1]: https://www.w3.org/TR/webvtt1/

**Reply by wc7086** [reply]
路走窄了，好像就没在github搜ass2vtt，刚发现这个库，似乎是能转换一些特效的，没试能转多少特效

https://github.com/civodulab/ass2vtt

**Reply by DogTorrent** [reply]
> 路走窄了，好像就没在github搜ass2vtt，刚发现这个库，似乎是能转换一些特效的，没试能转多少特效
> 
> https://github.com/civodulab/ass2vtt

我看看

**Reply by DogTorrent** [reply]
> 路走窄了，好像就没在github搜ass2vtt，刚发现这个库，似乎是能转换一些特效的，没试能转多少特效
> 
> https://github.com/civodulab/ass2vtt

我把[ass-to-vtt](https://github.com/jamiees2/ass-to-vtt)，[ass2vtt](https://github.com/civodulab/ass2vtt)都试了一下
你提到的ass2vtt需要自己折腾config，针对某个字幕的不同标签定义样式，其实相当于是个手动工具

ass-to-vtt效果倒是比subsrt好不少，但是在遇到特殊情况会直接error
![图片](https://user-images.githubusercontent.com/39661453/166004447-2d00a258-acce-4de9-9d8d-330bd75daabe.png)

我用的subsrt倒不会炸，就是遇到svg的时候会原封不动输出xml
![图片](https://user-images.githubusercontent.com/39661453/166004706-f2aea24c-dcbe-4c07-9715-2726de364e42.png)

其实最好还是ASS.js了，除了字体几乎没有毛病
![图片](https://user-images.githubusercontent.com/39661453/166006010-3f08207c-3245-458b-954d-907eb4ea1d04.png)

作为基准，这是本地播放器用XySubFilter输出的
![Snipaste_2022-04-30_01-56-19](https://user-images.githubusercontent.com/39661453/166006959-a715b822-5a35-44d1-844c-297b9496dba4.png)

**Reply by DogTorrent** [reply]
所以我觉得不应该想着怎么把ass转成更好的vtt，应该想办法把ass.js弄进去

**Reply by wc7086** [reply]
> 所以我觉得不应该想着怎么把ass转成更好的vtt，应该想办法把ass.js弄进去

确实，看来vtt是真的蛋疼。字体还挺好解决的，本地装有就行了，反正是自己看的。

**Reply by wc7086** [reply]
可惜我现在对WEB前端没啥兴趣，还是等着有空折腾ass.js好了。
